### ISSUE TITLE
'Range out' when read a color hash. 

### ISSUE THREAD

**Issue by wqmeng** [EtheaDev/SVGIconImageList#280]
```
function TColorConstList.GetColorValue(const ColorName: UTF8String; var Color: TColor32): Boolean;
var
//  Hash: Integer;
  Hash: cardinal;
  Item: PColorConstMapItem;
  LError: String;
begin
  try
    Hash := GetHash(ColorName);
  except
    on E: Exception do
      LError:= E.Message + E.StackTrace;
  end;
  Item := FBuckets[(Cardinal(Hash) and $7FFFFFFF) mod FMod];
  while (Item <> nil) and not IsSameAsciiUTF8String(Item.Data.ColorName, ColorName) do
    Item := Item.Next;
  if Item <> nil then
  begin
    Color := Item.Data.ColorValue;
    Result := True;
  end
  else
    Result := False;
end;
```

In the function, when read from a SVG file, the file has a fill color 'red', then I get error of 'range out', the matter is the function GetHash return a cardinal value, when I debug the 'red' color it return value of 2557620812, which is a valid cardinal value but not a valid integer value.

So I change the
var
// Hash: Integer;
Hash: cardinal;

And the error gets fixed. I am not very sure if I am missing something else.
If you have other advice, please let me know.

Thank you very much, Great Job.

The SVG file is a ca flag.
ca.svg
`<svg xmlns="http://www.w3.org/2000/svg" width="1200" height="600" viewBox="0 0 9600 4800"><path fill="red" d="M0 0h2400l99 99h4602l99-99h2400v4800H7200l-99-99H2499l-99 99H0z"/><path fill="#fff" d="M2400 0h4800v4800H2400zm2490 4430-45-863a95 95 0 0 1 111-98l859 151-116-320a65 65 0 0 1 20-73l941-762-212-99a65 65 0 0 1-34-79l186-572-542 115a65 65 0 0 1-73-38l-105-247-423 454a65 65 0 0 1-111-57l204-1052-327 189a65 65 0 0 1-91-27l-332-652-332 652a65 65 0 0 1-91 27l-327-189 204 1052a65 65 0 0 1-111 57l-423-454-105 247a65 65 0 0 1-73 38l-542-115 186 572a65 65 0 0 1-34 79l-212 99 941 762a65 65 0 0 1 20 73l-116 320 859-151a95 95 0 0 1 111 98l-45 863z"/></svg>`

**Comment by carloBarazzetta** [comment]
I've forwarded your issue to Image32 project:
https://github.com/AngusJohnson/Image32/issues/109

**Comment by AngusJohnson** [comment]
> And the error gets fixed. I am not very sure if I am missing something else.

No, you are correct, the hash variable should be an integer, not a cardinal.
Nevertheless, you shouldn't be getting an error because the function is protected using `{$OVERFLOWCHECKS OFF}`.

**Comment by carloBarazzetta** [comment]
Fixed in 4.2.1 version
